ocaml/xapi/xapi_http.ml

```diff
@@ -44,6 +44,18 @@ open Http
 
 exception NoAuth
 
+let task_id_of_req (req: Http.Request.t) =
+  let all = Http.Request.(req.cookie @ req.query) in
+  if List.mem_assoc "task_id" all
+  then Some (Ref.of_string (List.assoc "task_id" all))
+  else None
+
+let subtask_of_req (req: Http.Request.t) =
```

Comment:
Could you please help to merge if there are no other comments? Thanks. @thomassa @gaborigloi